x/crosschain/keeper/cctx_orchestrator_validate_inbound.go

```diff
@@ -48,3 +56,44 @@ func (k Keeper) ValidateInbound(
 
 	return &cctx, nil
 }
+
+// CheckMigration checks if the sender is a TSS address and returns an error if it is.
+// If the sender is an older TSS address, this means that it is a migration transfer, and we do not need to treat this as a deposit.
+func (k Keeper) CheckMigration(ctx sdk.Context, msg *types.MsgVoteInbound) error {
+	additionalChains := k.GetAuthorityKeeper().GetAdditionalChainList(ctx)
+	// Ignore cctx originating from zeta chain/zevm for this check as there is no TSS in zeta chain
+	if chains.IsZetaChain(msg.SenderChainId, additionalChains) {
```

Comment:
As it is related to observer/signer set, we should rather check if `CctxGateway == CCTXGateway_observers`

Reply:
The gateway interface only is responsible for the outbound only `ValidateOutbound` and its resolved after we are validate the inbound . 
This check is related to the inbound , and should be rejected before we even reach the `validateOutbound` part .

Is there a reason you can think of when we would want to allow a migration transaction to be treated as an inbound, for a different gateway?

Reply:
CctxGateway tells how the inbound/outbound are validated
For example for CCTXGateway_ibc this check will also be out of scope as inbounds don't use the observer/signer to be validated

Reply:
e248ad00f22119f135ad413b0cee8b0c719ab985